homework/src/index.js

```diff
@@ -31,17 +31,66 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
+    const root = document.getElementById("root");
+    const div = createAndAppend("div", root, { class: "container" });
+    const header = createAndAppend("p", div, {
+      text: "HYF Repositories",
+      class: "header"
+    });
+    const table = createAndAppend("table", root, { class: "table" });
+    const select = createAndAppend("select", header, { class: "select" });
+
+    const tbody = createAndAppend("tbody", table);
+    const tr = createAndAppend("tr", tbody);
+
+    createAndAppend("td", tr, tbody, { text: "Repository: ", class: "label" });
+    const repository = createAndAppend("a", tbody);
+    const description = createAndAppend("tr", tbody);
+    const forks = createAndAppend("tr", tbody, { class: "label" });
+
+    const updated = createAndAppend("tr", tbody, { class: "label" });
+
+    fetchJSON(url, (error, repositories) => {
+      if (error) {
+        createAndAppend("div", root, {
+          text: error.message,
+          class: "alert-error"
+        });
       } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
       }
+
+      repositories.forEach((element, index) => {
+        createAndAppend("option", select, {
+          text: element.name,
+          value: index
+        });
+      });
+
+      function repositoryBox() {
+        const index = select.value;
+        repository.innerText = select[index].textContent;
+        repository.setAttribute("href", repositories[index].html_url);
+        repository.setAttribute("target", "_blank");
+
+        forks.innerHTML = "Forks: " + repositories[index].forks;
+        description.innerHTML =
```

Comment:
Since `description` refers to a `tr` element, its `innerHTML` can only be `td` or `tr` elements. You can't/shouldn't set its `innerHTML` to a text string.